CMakeLists.txt

```diff
@@ -590,8 +590,19 @@ add_custom_command(
 add_custom_target(tests-jar DEPENDS ${TESTS_JAR})
 set_property(TARGET tests-jar PROPERTY JAR_FILE ${TESTS_JAR})
 
+# These flags are necessary in Java17+ in order for certain unit tests to
+# perform deep reflection on nonpublic members
+if (Java_VERSION_MAJOR VERSION_GREATER_EQUAL 17)
+  set(ADD_OPENS
```

Comment:
Let's rename this to `TEST_ADD_OPENS` so that it is even more clear that this is only to be used for tests.